python/paddle/fluid/layers/detection.py

```diff
@@ -1487,14 +1468,23 @@ def _is_list_or_tuple_and_equal(data, length, err_info):
     else:
         reshaped_boxes = []
         reshaped_vars = []
+
+        def _flatten_(v):
+            out = nn.reshape(x=v, shape=(-1, v.shape[-1]), inplace=True)
+            return out
+
         for i in range(len(box_results)):
-            reshaped_boxes.append(_reshape_with_axis_(box_results[i], axis=3))
-            reshaped_vars.append(_reshape_with_axis_(var_results[i], axis=3))
+            reshaped_boxes.append(_flatten_(box_results[i]))
+            reshaped_vars.append(_flatten_(var_results[i]))
 
         box = tensor.concat(reshaped_boxes)
         var = tensor.concat(reshaped_vars)
         mbox_locs_concat = tensor.concat(mbox_locs, axis=1)
+        mbox_locs_concat = nn.reshape(
+            mbox_locs_concat, shape=[0L, -1L, 4L], inplace=True)
```

Comment:
It seems that `0L` is not valid syntax for Python 3+. Just use `[0, -1, 4]` is ok.

Reply:
Done. Thanks!